src/cio_http_server.c

```diff
@@ -386,39 +402,48 @@ static void handle_accept(struct cio_server_socket *ss, void *handler_context, e
 	cio_buffered_stream_init(&client->bs, socket->get_io_stream(socket));
 
 	err = cio_timer_init(&client->read_timer, server->loop, NULL);
-	if (unlikely(err != cio_success)) {
-		if (server->error_cb != NULL) {
-			server->error_cb(server);
-		}
+	if (unlikely(err != CIO_SUCCESS)) {
+		goto timer_err;
+	}
 
-		client->bs.close(&client->bs);
-		return;
+	err = client->read_timer.expires_from_now(&client->read_timer, server->read_timeout_ns, client_timeout_handler, client);
+	if (unlikely(err != CIO_SUCCESS)) {
+		goto timer_err;
 	}
 
-	client->read_timer.expires_from_now(&client->read_timer, server->read_timeout_ns, client_timeout_handler, client);
 	client->finish_func = finish_request_line;
 	client->bs.read_until(&client->bs, &client->rb, CIO_CRLF, parse, client);
+
+	return;
+
+timer_err:
+	if (server->error_cb != NULL) {
+		server->error_cb(server);
+	}
+
+	client->bs.close(&client->bs);
+	return;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this redundant jump. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3AS3626)